code/app.py

```diff
@@ -1,381 +1,7 @@
-            "created": response.created,
-            "object": response.object,
-            "choices": [
-                {
-                    "messages": [
-                        {
-                            "role": "assistant",
-                            "content": response.choices[0].message.content,
-                        }
-                    ]
-                }
-            ],
-        }
-
-        return jsonify(response_obj), 200
-    else:
-        return Response(
-            stream_without_data(response), mimetype="application/json-lines"
-        )
-
-
-@app.route("/api/conversation/azure_byod", methods=["POST"])
-def conversation_azure_byod():
-    try:
-        if env_helper.should_use_data():
-            return conversation_with_data(request)
-        else:
-            return conversation_without_data(request)
-    except Exception as e:
-        errorMessage = str(e)
-        logging.exception(f"Exception in /api/conversation/azure_byod | {errorMessage}")
-        return (
-            jsonify(
-                {
-                    "error": "Exception in /api/conversation/azure_byod. See log for more details."
-                }
-            ),
-            500,
-        )
-
-
-def get_message_orchestrator():
-    from backend.batch.utilities.helpers.OrchestratorHelper import Orchestrator
-
-    return Orchestrator()
-
-
-def get_orchestrator_config():
-    from backend.batch.utilities.helpers.ConfigHelper import ConfigHelper
-
-    return ConfigHelper.get_active_config_or_default().orchestrator
-
-
-@app.route("/api/conversation/custom", methods=["POST"])
-def conversation_custom():
-    message_orchestrator = get_message_orchestrator()
-
-    try:
-        user_message = request.json["messages"][-1]["content"]
-        conversation_id = request.json["conversation_id"]
-        user_assistant_messages = list(
-            filter(
-                lambda x: x["role"] in ("user", "assistant"),
-                request.json["messages"][0:-1],
-            )
-        )
-
-        messages = message_orchestrator.handle_message(
-            user_message=user_message,
-            chat_history=user_assistant_messages,
-            conversation_id=conversation_id,
-            orchestrator=get_orchestrator_config(),
-        )
-
-        response_obj = {
-            "id": "response.id",
-            "model": env_helper.AZURE_OPENAI_MODEL,
-            "created": "response.created",
-            "object": "response.object",
-            "choices": [{"messages": messages}],
-        }
-
-        return jsonify(response_obj), 200
-
-    except Exception as e:
-        errorMessage = str(e)
-        logging.exception(f"Exception in /api/conversation/custom | {errorMessage}")
-        return (
-            jsonify(
-                {
-                    "error": "Exception in /api/conversation/custom. See log for more details."
-                }
-            ),
-            500,
-        )
-
+app = create_app()
```

Comment:
I'm not sure how this fixes the EnvHelper() side effect. This is moving the EnvHelper() call into create_app(), which is invoked here anyway. 

What am I missing?

Reply:
I believe the difference is that we're now importing a function in tests:
`from create_app import create_app`
Instead of a variable:
`from app import app`

Previously, just importing the app would invoke the `EnvHelper` to load all vars. Now I believe it will only invoke the `EnvHelper` when the `create_app` function is called, giving us much more control. Do correct me if I've misunderstood though @cecheta!

Reply:
Awesome, thanks for the explanation!

Reply:
Exactly, thanks @adamdougal 😄